Apex_Unit_Testing/Apex_Language_Test Case/Apex1 Test Case/MapStudentTest.cls

```diff
@@ -0,0 +1,42 @@
+@isTest
+public class MapStudentTest 
+{
+    @testsetup
+   	static void setupFOrClass() 
+    {
+        Class__c cls =new Class__c(Name='testclass',MaxSize__c=20);
+        insert cls;
+        List<Student__c> studentlist =new  List<Student__c>();
+        For(Integer i=0; i<5; i++ ) 
+        {
+            studentlist.add(new Student__c(Name='test',LastName__c='student'+i,Class__c=cls.Id));
+        }
+        insert studentlist;
+    }
+       
+    @isTest
+    static void mapStudentTest()
+    {
+        Class__c cls= [SELECT Id FROM Class__c LIMIT 1];
+        List<Student__c> studlist = MapStudent.getAllStudentByClassId(cls.Id);
+        List<Student__c> result = [SELECT Id FROM Student__c WHERE Class__c = :cls.Id];
+        System.assertEquals(result.size(), Studlist.size());
+    }
```

Comment:
this method is considered as positive flow. let's talk about some negative ones
1. What if I pass `null` as class Id?
2. What if the class Id I have provided doesn't have any student?